lib/Deprecated.js

```diff
@@ -64,16 +64,40 @@ class iPgm {
     * @param {string | array} data
     * @param {string} [type]
     * @param {object} options
-    * @param {*} inDs
     */
-  addParam(data, type, options, inDs) {
-    if (!Array.isArray(data) && !type) {
-      iPgmDeprecate('defaulting parameter type to 1024a has been deprecated. You should specify a type instead.');
-      // eslint-disable-next-line no-param-reassign
-      type = '1024a';
+  addParam(data, type, options) {
```

Comment:
Let's simplify this down to just taking an object. We can expand from there.

Reply:
This is in deprecated `iPgm.addParam` in this function we generate a object and call `ProgramCall.addParam` with that object.

Reply:
we kept the old signature here to maintain compatibility.

Reply:
We maintain compat with iPgm. We don't need to maintain compat with ProgramCall, because it never existed prior to 1.0. We can always add more options later.

Reply:
I agree with your statement. 

[Line 68](https://github.com/IBM/nodejs-itoolkit/pull/139/files/5f929ed8648b824a0af4d60a7d10afbf906da36c#diff-6c66e765bbe3b27f1ce21776dd217084R68) in `Deprecated.js` refrenced in the [comment](https://github.com/IBM/nodejs-itoolkit/pull/139/files/5f929ed8648b824a0af4d60a7d10afbf906da36c#r394640338) is for iPgm though. If we change this to accept an object then we would break comparability for `iPgm` right?

In [ProgramCall.addParam](https://github.com/IBM/nodejs-itoolkit/pull/139/files/5f929ed8648b824a0af4d60a7d10afbf906da36c#diff-062b945422d948ff48bd40d77ed2de0fR122) we only accept an object parameter already.

Reply:
No, iPgm will need to be adjusted to map its arguments to ProgramCall's

Reply:
Yes the parameters are being mapped Lines 69 -97 below and the ProgramCall.addParam is called with the mapped object on [line 100](https://github.com/IBM/nodejs-itoolkit/pull/139/files/5f929ed8648b824a0af4d60a7d10afbf906da36c#diff-6c66e765bbe3b27f1ce21776dd217084R100)

Reply:
Ooops, I was looking at the wrong function.